### PR TITLE
Add demo and README for demo. Minor edit of main's README to add data description

### PR DESCRIPTION
**Pull request by nguyenntt97**
- Add new demo folder with README in `/demo`—see [new Demo's README](https://github.com/onurem/GSE---TEAM-A/tree/feat/api/demo)
- Add more data description in repo page's README—see [updated README](https://github.com/onurem/GSE---TEAM-A/tree/feat/api#data-description)
- Update projects for frontend and backend for the demo—demo available at [Hateless Heroku app](https://hateless.herokuapp.com/)

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Two main components for project:
+
+- Backend API
+- Frontend web application
+
+**Note:**
+
+- Frontend web app is built and delivered by backend api at root
+- To run the demo on your local, please re-create virtual environment with dependency listed in `web_api/requirements.txt` then execute the `wsgi.py` with the venv's Python
+- To run the frontend only, change directory to `web_app/hateless` then run `npm run start`
```
Comment (JohnnyDoesStuff) [inline demo/README.md:42]: Which npm dependencies are required to run this? It would help me to have a short "Getting Started" section which briefly describes the steps i need to do so that i can run this application.

File: demo/web_api/app.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+vtrizers = {}
+load_model()
+
+@app.route('/', methods=['GET'])
+def index():
+    return render_template('index.html')
+
+
+@app.route('/<version>/predict', methods=['POST'])
+def predict(version):
```
Comment (JohnnyDoesStuff) [inline demo/web_api/app.py:45]: this function could use the Predictor class.  (This can also be done in an own pull request)

**Comment by JohnnyDoesStuff** [comment]
the dataset.csv file should not be versionated since we can retrieve the data from other sources and i doubt that we really want to change it.

File: demo/web_api/static/static/js/main.b7f6f124.js.LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/*
```
Comment (JohnnyDoesStuff) [inline demo/web_api/static/static/js/main.b7f6f124.js.LICENSE.txt:1]: Is this license file required by some class or was it just part of a template? (In the latter case i think we can remove it)

File: demo/web_app/hateless/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (JohnnyDoesStuff) [inline demo/web_app/hateless/README.md]: I think this file can be removed (Can be postponed to an own pull request)

**Comment by JohnnyDoesStuff** [comment]
the package.lock.json should also not be versionated as far as i know (I'm writing this comment here since github does not want to open it in the review-view)

**Comment by JohnnyDoesStuff** [comment]
about the _Data description_ section in the README.md: i think we should remove it again since we're currently using multiple datasets which have been described by a different section which has been merged previously (also: Why does github consider this file binary?)

**Review by JohnnyDoesStuff — CHANGES_REQUESTED** [review]
General points besides my other comments:
- some paths still have to be adapted (e.g. running "python -m pytest" from the repository's root does no longer work --> please fix that)
- just a hint for the future: smaller pull requests are easier to review / change / merge

File: demo/README.md (hunk 1 above)
Comment (nguyenntt97) [reply demo/README.md:42]: The npm dependencies can be found in [package.json file](https://github.com/onurem/GSE---TEAM-A/blob/feat/api/demo/web_app/hateless/package.json)

Basically just React and MUI (for material UI components)

**Comment by nguyenntt97** [comment]
> the dataset.csv file should not be versionated since we can retrieve the data from other sources and i doubt that we really want to change it.

Did you mean [this dataset.csv](https://github.com/onurem/GSE---TEAM-A/blob/feat/api/demo/web_api/data/dataset.csv)?

Not sure if it's what you meant though. The only version postfix I found is for the version control purpose of the models (a pair of vectorizers and classifiers).

File: demo/web_api/app.py (hunk 2 above)
Comment (nguyenntt97) [reply demo/web_api/app.py:45]: Great idea, I'll take a look at the Predictor class

File: demo/web_api/static/static/js/main.b7f6f124.js.LICENSE.txt (hunk 3 above)
Comment (nguyenntt97) [reply demo/web_api/static/static/js/main.b7f6f124.js.LICENSE.txt:1]: it was included by default in the build process of my React project, it included all licenses of the dependencies used in a React project. I'll check if there is any option to turn it off.

File: demo/README.md (hunk 1 above)
Comment (nguyenntt97) [reply demo/README.md:42]: I'll add the `Getting Started` section tonight

**Comment by nguyenntt97** [comment]
@JohnnyDoesStuff please check the changes I've just committed.

3 latest commits:

1. Add `getting started` for React FE project
2. Remove default `README.md` file generated by React
3. Remove `dataset description` from the root `README.md`

- I'll save the `Predictor` change in the API to the next PR

Though, I need some clarifications:

- Regarding the versioned dataset that you mentioned. It was there for my self-trained model, after I integrated your model into the API, then it would be removed. Is it ok?

**Comment by JohnnyDoesStuff** [comment]
> @JohnnyDoesStuff please check the changes I've just committed.
> 
> 3 latest commits:
> 
> 1. Add `getting started` for React FE project
> 2. Remove default `README.md` file generated by React
> 3. Remove `dataset description` from the root `README.md`
> 
> * I'll save the `Predictor` change in the API to the next PR
> 
> Though, I need some clarifications:
> 
> * Regarding the versioned dataset that you mentioned. It was there for my self-trained model, after I integrated your model into the API, then it would be removed. Is it ok?

1. I still got trouble to run it via npm:
```
> hateless@0.1.0 start
> CHOKIDAR_USEPOLLING=true react-scripts start

'CHOKIDAR_USEPOLLING' is not recognized as an internal or external command,
operable program or batch file.
```
2. good
3. good

- Predictor is ok
- about the dataset (demo/web_api/data/dataset.csv): yes, i think, we can remove it
- "python -m pytest" is still not working

File: demo/web_api/app.py (hunk 2 above)
Comment (JohnnyDoesStuff) [reply demo/web_api/app.py:45]: as discussed: has been postponed to another pull request

**Review by JohnnyDoesStuff — APPROVED** [review]
